### PR TITLE
Touch up help page visually and clean up content

### LINKED ISSUE #76: help (1.4 ver)
im not sure if you are still working on this

### PR DESCRIPTION
**Pull request by Shantaa**
Closes #76

### CODE REVIEW

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 1]
```diff
@@ -16,7 +29,25 @@ public class HelpWindow extends UiPart<Stage> {
     private static final String FXML = "HelpWindow.fxml";
 
     @FXML
-    private TextArea helpContent;
+    private TextFlow helpContentFlow;
```
Comment (meatyturtle) [inline src/main/java/seedu/address/ui/HelpWindow.java:32]: Really detailed and extensive research into FXML elements is evident. Well done, Shane!

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 2]
```diff
@@ -81,117 +112,278 @@ public void focus() {
...
-    private void setHelpContent() {
-        String userGuideContent = """
-            Financial Assurance Revolutionary Telemarketer (FART) is a **desktop app for managing contacts just for you
-             Financial Advisors, optimized for use via a Command Line Interface** (CLI)
-             while still having the benefits of a Graphical User Interface (GUI).
-             If you can type fast, FART can get your contact management tasks done faster than traditional GUI apps.
-
-            ## Quick start
-
+    private void initializeContents() {
```
Comment (meatyturtle) [inline src/main/java/seedu/address/ui/HelpWindow.java:117]: Good job abstracting contents into array elements

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 3]
```diff
@@ -81,117 +112,278 @@ public void focus() {
...
+                """);
+
+        contentMap.put("Clearing All Entries", """
+                Clearing All Entries\n
+                Clears all entries from the FART book.\n
+                Format: `clear`\n
+                This command will remove all contacts from the FART book.
+                """);
+
+        contentMap.put("Marking a Person as Paid/Unpaid", """
```
Comment (meatyturtle) [inline src/main/java/seedu/address/ui/HelpWindow.java:230]: I think it's better to be consistent and choose to leave either 1 or 2 lines in between each contentMap.put

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 4]
```diff
@@ -81,117 +112,278 @@ public void focus() {
...
+                | Paid   | `paid INDEX` | `paid 3` |
+                | Unpaid | `unpaid INDEX` | `unpaid 3` |
+                """);
+
+        initializeCommandSummaryData();
+
+        updateTextFlow(contentMap.get("Introduction"));
+
+        tocListView.getSelectionModel().selectedItemProperty().addListener((observable, oldValue, newValue) -> {
+            if (newValue != null) {
```
Comment (meatyturtle) [inline src/main/java/seedu/address/ui/HelpWindow.java:278]: Is there another way to circumvent this somewhat "deep" nesting?

File: src/main/resources/view/HelpWindow.css [hunk 5]
```diff
@@ -1,19 +1,34 @@
...
+#helpTitle {
+    -fx-font-size: 24px;
+    -fx-font-weight: bold;
+    -fx-text-fill: #2c3e50;
+    -fx-padding: 10 0 20 0;
 }
 
-#copyButton:hover {
-    -fx-background-color: gray;
+#helpContent {
```
Comment (meatyturtle) [inline src/main/resources/view/HelpWindow.css:12]: love the new styles!

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 6]
```diff
@@ -81,117 +112,278 @@ public void focus() {
...
-        helpContent.setWrapText(true);
-        helpContent.setEditable(false);
+    /**
+     * Opens a hyperlink in the default browser.
+     */
+    private void openHyperlink(String url) {
+        try {
+            if (Desktop.isDesktopSupported()) {
+                Desktop.getDesktop().browse(new URI(url));
+            } else {
```
Comment (meatyturtle) [inline src/main/java/seedu/address/ui/HelpWindow.java:381]: As per v1.3 I believe the else branch will always be taken? is there a need for the if branch in this case

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 7]
```diff
@@ -81,117 +112,278 @@ public void focus() {
...
-            ### Saving the data
+        if (content.contains("Quick start")) {
+            Text beforeLink1 = new Text("\n1. Ensure you have Java ");
+            Hyperlink javaLink = new Hyperlink("https://www.oracle.com/java/technologies/downloads/#java17");
+            javaLink.setOnAction(e -> openHyperlink("https://www.oracle.com/java/technologies/downloads/#java17"));
 
-            FART data is saved in the hard disk automatically after any command that changes the data.
-             There is no need to save manually.
+            Text afterLink1 = new Text("\n\n2. Download the latest ");
+            Hyperlink jarLink = new Hyperlink("https://github.com/se-edu/addressbook-level3/releases");
```
Comment (meatyturtle) [inline src/main/java/seedu/address/ui/HelpWindow.java:339]: I believe the link should point to FART instead, not AB3 :/

File: src/main/resources/view/HelpWindow.fxml [hunk 8]
```diff
@@ -19,19 +25,36 @@
         <URL value="@HelpWindow.css" />
       </stylesheets>
 
-      <!-- VBox to hold the content with Insets for padding -->
-      <VBox alignment="CENTER">
-        <padding>
-          <Insets bottom="10.0" left="10.0" right="10.0" top="10.0" />
-        </padding>
+      <SplitPane orientation="HORIZONTAL" dividerPositions="0.4" prefWidth="800" prefHeight="500">
```
Comment (meatyturtle) [inline src/main/resources/view/HelpWindow.fxml:28]: fire

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 9]
```diff
@@ -81,117 +112,278 @@ public void focus() {
...
+        helpContentFlow.getChildren().clear();
 
-            ### Exiting the program : `exit`
+        String[] parts = content.split("\n", 2);
+        String titleText = parts[0];
+        String bodyText = parts.length > 1 ? parts[1] : "";
 
-            Exits the program.
+        Text title = new Text(titleText + "\n");
+        title.setStyle("-fx-font-weight: bold; -fx-font-size: 18px;");
```
Comment (meatyturtle) [inline src/main/java/seedu/address/ui/HelpWindow.java:330]: CMIIW, but I think you can abstract the styles out into the css file or the fxml file to keep the java code clean

**Review by meatyturtle — COMMENTED** [review]
Nice work, just a few minor nitpicks but LGTM otherwise.

File: src/main/java/seedu/address/ui/HelpWindow.java (hunk 6 above)
Comment (Shantaa) [reply src/main/java/seedu/address/ui/HelpWindow.java:423]: Could I clarify what you mean by this, as this is meant to check if the user's device is able to support desktop browsing to access hyperlinks?

File: src/main/java/seedu/address/ui/HelpWindow.java (hunk 7 above)
Comment (Shantaa) [reply src/main/java/seedu/address/ui/HelpWindow.java:381]: good catch, rectified accordingly thanks!

File: src/main/resources/view/HelpWindow.fxml (hunk 8 above)
Comment (Shantaa) [reply src/main/resources/view/HelpWindow.fxml:28]: appreciate your vote of confidence

File: src/main/java/seedu/address/ui/HelpWindow.java (hunk 6 above)
Comment (meatyturtle) [reply src/main/java/seedu/address/ui/HelpWindow.java:423]: ah i see cause i remember there is an evolved version of this called AB4 which supports browsing from within the app itself
but if you wrote this code, then lgtm

File: src/main/java/seedu/address/ui/HelpWindow.java (hunk 4 above)
Comment (Shantaa) [reply src/main/java/seedu/address/ui/HelpWindow.java:110]: happy to inform you that the deep nesting has been circumvented

**Pull request merged**